packages/at_client/lib/src/service/notification_service_impl.dart

```diff
@@ -102,33 +112,47 @@ class NotificationServiceImpl
 
   @visibleForTesting
   Future<int?> getLastNotificationTime() async {
-    if (_atClientManager.atClient.getPreferences()!.fetchOfflineNotifications == false) {
+    if (_atClientManager.atClient.getPreferences()!.fetchOfflineNotifications ==
+        false) {
       // fetchOfflineNotifications == false means issue `monitor` command without a lastNotificationTime
       // which will result in the server not sending any previously received notifications
       return null;
     }
 
     // fetchOfflineNotifications == true (the default) means we want all notifications since the last one we received
     // We keep track of the last notification id in the client-side key store
-    var lastNotificationKeyStr =
-        '$notificationIdKey.${_atClient.getPreferences()!.namespace}${_atClient.getCurrentAtSign()}';
-    var atKey = AtKey.fromString(lastNotificationKeyStr);
+    // Check if the new key (local:lastNotificationReceived@alice) is available in the keystore.
+    // If yes, fetch the value;
+    AtValue? atValue;
     if (_atClient
         .getLocalSecondary()!
         .keyStore!
-        .isKeyExists(lastNotificationKeyStr)) {
-      AtValue? atValue;
-      try {
-        atValue = await _atClient.get(atKey);
-      } on Exception catch (e) {
-        _logger
-            .severe('Exception in getting last notification id: ${e.toString}');
-      }
-      if (atValue != null && atValue.value != null) {
-        _logger.finer('json from hive: ${atValue.value}');
-        return jsonDecode(atValue.value)['epochMillis'];
+        .isKeyExists(lastReceivedNotification.toString())) {
+      atValue = await _atClient.get(lastReceivedNotification);
+    }
+    // If new key does not exist or value is null, check for the old key (_latestNotificationId@alice)
```

Comment:
in the comment..should be _latestNotificationIdv2@alice ?

Reply:
Updated the comment.